tools/src/main/java/org/apache/kafka/tools/SchemaChecker/MetadataSchemaChecker.java

```diff
@@ -0,0 +1,359 @@
+                    if (line.charAt(0) == '{') {
+                        print = true;
+                    }
+                    if (print && !line.contains("//")) {
+                        content.append(line);
+                    }
+                }
+                localContent.add(content.toString());
+            }
+
+            List<String> gitContent = GetDataFromGit();
+            if (localContent.size() != gitContent.size()) {
+                throw new IllegalStateException("missing schemas");
+            }
+            for(int i = 0; i < localContent.size(); i++) {
+                if (Objects.equals(localContent.get(i), gitContent.get(i))) {
+                    continue;
+                }
+
+                ObjectMapper objectMapper = new ObjectMapper();
+                JsonNode jsonNode1 = objectMapper.readTree(gitContent.get(i));
+                JsonNode jsonNode2 = objectMapper.readTree(localContent.get(i));
+
+                checkApiTypeVersions(jsonNode1, jsonNode2);
+                parser((ArrayNode) jsonNode1.get("fields"), (ArrayNode) jsonNode2.get("fields"));
+            }
+
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static List<String> GetDataFromGit() throws IOException, GitAPIException {
+        List<String> gitSchemas = new ArrayList<>();
+
+        Git git = Git.open(new File(System.getProperty("user.dir") + "/.git"));
+        Repository repository = git.getRepository();
+        Ref head = git.getRepository().getRefDatabase().firstExactRef("refs/heads/trunk");
+
+        try (RevWalk revWalk = new RevWalk(repository)) {
+            RevCommit commit = revWalk.parseCommit(head.getObjectId());
+            RevTree tree = commit.getTree();
+            for (String fileName : filesCheckMetadata) {
+                StringBuilder stringBuilder = new StringBuilder();
+                try (TreeWalk treeWalk = new TreeWalk(repository)) {
+                    treeWalk.addTree(tree);
+                    treeWalk.setRecursive(true);
+                    treeWalk.setFilter(PathFilter.create("metadata/src/main/resources/common/metadata/" + fileName));
+                    if (!treeWalk.next()) {
+                        throw new IllegalStateException("Did not find expected file /metadata/src/main/resources/common/metadata/" + fileName);
+                    }
+                    ObjectId objectId = treeWalk.getObjectId(0);
+                    ObjectLoader loader = repository.open(objectId);
+
+                    String content = new String(loader.getBytes());
+                    String[] lines = content.split("\n");
+                    boolean print = false;
+                    for (int i = 15; i < lines.length; i++) {
+                        if (lines[i].charAt(0) == '{') {
+                            print = true;
+                        }
+                        if (print && !lines[i].contains("//")) {
```

Comment:
It's not enough for the line to contain //. It needs to start with // (after whitespace has been trimmed)